src/Icicle/Source/Transform/Desugar.hs

```diff
@@ -322,7 +323,10 @@ data Tree a n x
          [(Constructor, Tree a n x)] (Tree a n x)
  deriving (Functor, Foldable, Traversable, Show)
 
-
+instance Applicative (Tree a n) where
+  pure = return
```

Comment:
fwiw, this doesn't work according to the docs, not sure what the effect is, but they say you're supposed to move the `return` implementation here and make `return = pure`